docs/content/post/developer-documentation.md

```diff
@@ -0,0 +1,256 @@
+target     prot opt source               destination
+SNAT       all  --  anywhere             anywhere             to:192.168.10.99
+
+```
+
+## Using quay.io/openshift image
+
+In order to deploy this into a cluster, you'll have to either compile the
+inclusded `Dockerfile` or rely into the preexising images that are shipped into
+openshift's quay
+(https://quay.io/repository/openshift/origin-egress-router-cni).
+
+If deploying an openshift cluster, we're already deploying this image alongside
+CNO's multus' daemonsets, so you've got nothing to do. Only if you want to
+change the default image to yours, you'll have to override the
+`EGRESS_ROUTER_CNI_IMAGE` value with the reference to your image.
+
+If you don't want to use CNO at all, you can still install the egress router
+cni by manually copying it to the CNI binary path, which should be
+`/var/lib/cni/bin/`
+
+### How to use the egress-router-cni
+
+Egress Router CNI aims to match openshift-sdn functionality, please refer to
+[this nice openshift blog article](https://www.openshift.com/blog/accessing-external-services-using-egress-router)
+for further details on it. In order to make things easier for the legacy router
+users, so far the `redirect` mode is supported in egress-router-cni.
+
+In openshift-sdn, the egress router was implemented by adding an annotation to
+allow a pod to request a macvlan interface. In order to avoid repeating this
+behavior in ovn-kubernetes, we'd be requesting such interface using multus to
+ensure feature-parity with openshift-sdn.
+
+Please take a look at the [enhancement
+proposal](https://github.com/openshift/enhancements/blob/master/enhancements/network/egress-router.md)
+if you want to get some more detailed info on this.
+
+Assumming that you have the cni plugin installed, you'd be able to create a
+NetworkAttachmentDefinition (NAD) with the CNI configuration alongside, just
+such as in the below example.
+
+> NOTE:
+> Multus is a requirement for egress-router-cni usage.
+
+The 'NetworkAttachmentDefinition' is used to setup the network attachment, i.e.
+secondary interface for the pod, There are two ways to configure the
+'NetworkAttachmentDefinition' as following:
+
+* NetworkAttachmentDefinition with json CNI config
+* NetworkAttachmentDefinition with CNI config file
+
+#### NetworkAttachmentDefinition with json CNI config
+
+```yaml
+---
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: egress-router
+spec:
+  config: '{
+    "cniVersion": "0.4.0",
+    "type": "egress-router",
+    "name": "egress-router",
+    "ip": {
+      "addresses": [
+        "10.200.16.0/24"
+        ],
+      "destinations": [
+        "10.100.3.0/32"
+      ],
+      "gateway": "192.168.10.200"
+      }
+    }'
+```
+
+This would create the additional network, which would be later used in the pod
+with the macvlan interface.
+
+Let's go create the pod!
+
+> NOTE:
+> A pod image with iptables is required.
```

Comment:
This is not true; egress-router-cni uses the system's iptables binaries, not anything from the pod image.